### PR TITLE
fix(text-field): Fix for input alignment in textfield with trailing icon

### LINKED ISSUE #4416: Filled text fields floating label doesn't align with trailing icon
### What browser(s) is this bug affecting?

Chome

### What OS are you using?

ChomeOS

### What are the steps to reproduce the bug?

Have a boxed input with a floating label and a trailing icon, input some text. The floated label will not align with the input.

### What is the expected behavior?

The floated label and the text will align.

### PR DESCRIPTION
**Pull request by abhiomkar**
Made changes to private mixin of `text-field/icon`: `mdc-text-field-icon-horizontal-position_` by adding a new argument to send default text field input padding for trailing icon variant.

In previous implementation it assumes the leading input padding and the icon position always equals, but it is not the case in filled variant since the icon position is 4px lesser (`12px`) in filled variant compared to outlined variant (`16px`).

**Before**

**After**

Fixes #4416

### CODE REVIEW

File: packages/mdc-textfield/icon/_mixins.scss [hunk 1]
```diff
@@ -42,7 +42,7 @@
     }
   } @else {
     .mdc-text-field__input {
-      @include mdc-rtl-reflexive-property(padding, $position /* left-value */, $padding  /* right-value */);
+      @include mdc-rtl-reflexive-property(padding, $input-padding /* left-value */, $padding  /* right-value */);
```
Comment (kfranqueiro) [inline packages/mdc-textfield/icon/_mixins.scss]: I'm a little bit confused as to why this change is only needed within the `else`?

Comment (abhiomkar) [reply packages/mdc-textfield/icon/_mixins.scss]: Used `$input-padding` to `left` position too to make it semantically correct. Icon's `position` value and text field's `input-padding` are same anyway.
